als-suggestions/shared/src/main/scala/org/mulesoft/als/suggestions/plugins/aml/UnionSuggestions.scala

```diff
@@ -21,4 +23,36 @@ trait UnionSuggestions {
       case d if d.id == id => d
     })
 
+  lazy val definedProperties: Set[String] = yPartBranch.brothersKeys
+
+  protected def hasAllDefinedProperties(value: NodeMapping, typeDiscriminatorName: Option[String]): Boolean =
+    definedProperties
+      .filterNot(typeDiscriminatorName.contains)
+      .forall(property => value.propertiesMapping().exists(_.name().value() == property))
+
+  protected def getUnionProperties(unionMapping: UnionNodeMapping): Seq[PropertyMapping] =
+    unionMapping
+      .objectRange()
+      .map(_.value())
+      .map(getDeclaredDomainElement)
+      .flatMap(getProperties(_, unionMapping.typeDiscriminatorName().option()))
+      // filter duplicates (same property defined in multiple ranges of the union)
+      .foldLeft(Seq[PropertyMapping]())(filterProperties)
+
+  protected def getProperties(nm: Option[DomainElement],
+                              typeDiscriminatorName: Option[String] = None): Seq[PropertyMapping] =
+    nm match {
+      case Some(value: UnionNodeMapping) => getUnionProperties(value)
+      case Some(value: NodeMapping) if hasAllDefinedProperties(value, typeDiscriminatorName) =>
+        value.propertiesMapping()
+      case _ => Seq.empty
+    }
+
+  protected def filterProperties(acc: Seq[PropertyMapping], prop: PropertyMapping): Seq[PropertyMapping] =
+    if (!acc.exists(p => isDuplicateOf(p, prop))) acc :+ prop else acc
```

Comment:
couldn't this be replaced by doing a `.distinct` or replacing the `Seq` by a `Set`?

Reply:
I was afraid of how Any implements the equality function for PropertyMapping. Still not sure, but seems to be working as expected, so I'll make the requested change

Reply:
My bad, not working as expected, missed a test. Will revert the change